lib/compress.js

```diff
@@ -1273,7 +1273,21 @@ merge(Compressor.prototype, {
         def(AST_Constant, return_false);
         def(AST_This, return_false);
 
+        var pure_regex = /[@#]pure/;
```

Comment:
I think the directive should be allowed to appear anywhere in the comment. The directive doesn't have to pretty - it's *uglify*, after all. ;-)

`/[@#]__PURE__/` should lower the odds of accidental comment collision.